File: src/main/java/com/weeth/domain/attendance/domain/service/AttendanceSaveService.java
package com.weeth.domain.attendance.domain.service;

import jakarta.transaction.Transactional;
import com.weeth.domain.attendance.domain.entity.Attendance;
import com.weeth.domain.attendance.domain.entity.Session;
import com.weeth.domain.attendance.domain.repository.AttendanceRepository;
import com.weeth.domain.user.domain.entity.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
@RequiredArgsConstructor
public class AttendanceSaveService {

    private final AttendanceRepository attendanceRepository;

    public void init(User user, List<Session> sessions) {
        if (sessions != null) {
            sessions.forEach(session -> {
                Attendance attendance = attendanceRepository.save(Attendance.Companion.create(session, user));
                user.add(attendance);
            });
        }
    }

    public void saveAll(List<User> userList, Session session) {
        List<Attendance> attendances = userList.stream()
                .map(user -> Attendance.Companion.create(session, user))
                .toList();

        attendanceRepository.saveAll(attendances);
    }
}

File: src/main/kotlin/com/weeth/domain/attendance/domain/service/AttendanceSaveService.kt
package com.weeth.domain.attendance.domain.service

import com.weeth.domain.attendance.domain.entity.Attendance
import com.weeth.domain.attendance.domain.repository.AttendanceRepository
import com.weeth.domain.schedule.domain.entity.Meeting
import com.weeth.domain.user.domain.entity.User
import org.springframework.stereotype.Service

@Service
class AttendanceSaveService(
    private val attendanceRepository: AttendanceRepository,
) {
    fun init(
        user: User,
        meetings: List<Meeting>?,
    ) {
        meetings?.forEach { meeting ->
            attendanceRepository.save(Attendance(meeting, user))
        }
    }

    fun saveAll(
        userList: List<User>,
        meeting: Meeting,
    ) {
        val attendances = userList.map { user -> Attendance(meeting, user) }
        attendanceRepository.saveAll(attendances)
    }
}
